### PR TITLE
Features/rfh2 support

### PR DESCRIPTION
**Pull request by boffman**
Basic support for reading and writing gzip compressed messages using RFH2 header

### CODE REVIEW

File: grizzly/users/messagequeue.py [hunk 1]
```diff
@@ -197,14 +213,23 @@ def __init__(self, environment: Environment, *args: Tuple[Any, ...], **kwargs: D
 
         auth_context = self._context.get('auth', {})
         username = auth_context.get('username', None)
+        message_context = self._context.get('message', {})
+        header_type = message_context.get('header_type', None)
+        if header_type:
+            header_type = header_type.lower()
+            if header_type not in ['rfh2', 'none']:
```
Comment (mgor) [inline grizzly/users/messagequeue.py:220]: where is `header_type` set to `"none"` ? 
line 218, if value is the default (as specified on line 158) would be `None` (`NoneType`), would not get down to this check? my understanding is that `None` would be the same as `"none"`.

File: grizzly/users/messagequeue.py [hunk 2]
```diff
@@ -197,14 +213,23 @@ def __init__(self, environment: Environment, *args: Tuple[Any, ...], **kwargs: D
...
+                raise ValueError(f'{self.__class__.__name__} unsupported value for header_type: "{message_context["header_type"]}", supported ones are "None" and "RFH2"')
```
Comment (mgor) [inline grizzly/users/messagequeue.py:221]: `{message_context["header_type"]}` could be `header_type` in the f-string.

File: grizzly_extras/async_message/mq/rfh2.py [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+            return False
+
+    def __init__(self, message: bytes) -> None:
+        self._parse_header(message)
+        self._parse_name_values()
+
+    def _parse_header(self, message: bytes) -> None:
+        header_length = 36
+        try:
+            parts = unpack('<4c l l l l 8c l l', message[0:header_length])
```
Comment (mgor) [inline grizzly_extras/async_message/mq/rfh2.py:34]: magic :D

File: grizzly_extras/async_message/mq/rfh2.py [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+            self.struc_id = b''.join(list(parts[0:4])).decode('ascii')
+            self.version = parts[4]
+            self.struc_length = parts[5]
+            self.encoding = parts[6]
+            self.charset = parts[7]
+            self.fmt = b''.join(list(parts[8:16])).decode('ascii')
+            self.flags = parts[16]
+            self.name_value_ccsid = parts[17]
+            self.name_values = message[header_length:self.struc_length]
```
Comment (mgor) [inline grizzly_extras/async_message/mq/rfh2.py:43]: does everything between line 35-43 really need to be class variables?

File: grizzly_extras/async_message/mq/rfh2.py [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+    def _build_header(self) -> None:
+        header_length = 36
+        self.struc_id = 'RFH '
+        self.version = 2
+        self.struc_length = header_length + len(self.name_values)
+        self.encoding = 546
+        self.charset = 1208
+        self.fmt = ' ' * 8
+        self.flags = 0
+        self.name_value_ccsid = 1208
```
Comment (mgor) [inline grizzly_extras/async_message/mq/rfh2.py:141]: does everything between line 134-141 need to be class variables?
most of them looks like they can be constants, and none of them seems to be used outside of `_builder_header`.

File: grizzly_extras/async_message/mq/__init__.py [hunk 6]
```diff
@@ -2,18 +2,19 @@
...
+from .. import (
     AsyncMessageRequest,
     AsyncMessageResponse,
     AsyncMessageError,
     AsyncMessageRequestHandler,
     AsyncMessageHandler,
-    register,
+    register
 )
+from .rfh2 import Rfh2Decoder, Rfh2Encoder
```
Comment (mgor) [inline grizzly_extras/async_message/mq/__init__.py:18]:
```python
__all__ = [
 'Rfh2Decoder',
 'Rfh2Encoder',
]
```

so its possible to import them via `grizzly_extras.async_message.mq` without needing to specify `rfh2` in the namespace.

**Review by mgor — CHANGES_REQUESTED** [review]
must have support in `grizzly.tasks.clients.messagequeue` and should have support in `grizzly.testdata.variables.messagequeue`.

File: grizzly/users/messagequeue.py (hunk 1 above)
Comment (boffman) [reply grizzly/users/messagequeue.py:220]: The message.header_type context variable can be set to either "RFH2" or "none" (or be left out), where "none" would mean the same as not specifying header_type at all.

File: grizzly/users/messagequeue.py (hunk 2 above)
Comment (boffman) [reply grizzly/users/messagequeue.py:221]: check!

File: grizzly_extras/async_message/mq/rfh2.py (hunk 4 above)
Comment (boffman) [reply grizzly_extras/async_message/mq/rfh2.py:43]: Nope! :)

File: grizzly_extras/async_message/mq/rfh2.py (hunk 5 above)
Comment (boffman) [reply grizzly_extras/async_message/mq/rfh2.py:141]: Nope! :)

File: grizzly/users/messagequeue.py (hunk 1 above)
Comment (mgor) [reply grizzly/users/messagequeue.py:220]: this.

**Review by mgor — CHANGES_REQUESTED** [review]
one unresolved comment :)

good work!

**Review by mgor — APPROVED** [review]

**Pull request merged**